Detectors/MUON/MID/Workflow/src/MaskHandlerSpec.cxx

```diff
@@ -0,0 +1,129 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file   MID/Workflow/src/MaskHandlerSpec.cxx
+/// \brief  Processor to handle the masks
+/// \author Diego Stocco <Diego.Stocco at cern.ch>
+/// \date   13 August 2021
+
+#include "MIDWorkflow/MaskHandlerSpec.h"
+
+#include <array>
+#include <vector>
+#include <chrono>
+#include <gsl/gsl>
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/ControlService.h"
+#include "Framework/InputSpec.h"
+#include "Framework/Logger.h"
+#include "Framework/Task.h"
+#include "DataFormatsMID/ColumnData.h"
+#include "DataFormatsMID/ROFRecord.h"
+#include "MIDFiltering/ChannelMasksHandler.h"
+
+namespace of = o2::framework;
+
+namespace o2
+{
+namespace mid
+{
+
+class MaskHandlerDeviceDPL
+{
+ public:
+  void init(o2::framework::InitContext& ic)
+  {
+
+    auto stop = [this]() {
+      printSummary();
+    };
+    ic.services().get<of::CallbackService>().set(of::CallbackService::Id::Stop, stop);
+  }
+
+  void printSummary()
+  {
+    std::string name = "calib";
+    for (auto& masks : mMasksHandlers) {
+      auto maskVec = masks.getMasks();
+      if (maskVec.empty()) {
+        LOG(INFO) << "No problematic digit found in " << name << " events";
+      } else {
+        LOG(INFO) << "Problematic digits found in " << name << " events. Corresponding masks:";
+        for (auto& mask : maskVec) {
+          LOG(INFO) << mask;
+        }
+      }
+      name = "FET";
+    }
+  }
+
+  // std::cout << "\nCorresponding boards masks:" << std::endl;
```

Comment:
I guess this commented out part can go, right ?

Reply:
Well, this code allowed indeed to show the mask as a Local Board mask. This makes it easier to apply the mask at DCS level. I'm therefore uncommenting it ;)
But ok, this is secondary since, in any case, in the end we will have to put the mask in the CCDB...